scalyr_agent/log_processing.py

```diff
@@ -1848,14 +1859,26 @@ def process_line(self, input_line):
 
         return input_line, modified_it
 
-    def add_redaction_rule(self, redaction_expression, replacement_text):
+    def add_redaction_rule(self, redaction_expression, replacement_text,
+                           hash_replace=False, hash_salt=''):
```

Comment:
Since you are now indicating, in the replacement string, whether or not to use a hash of the value rather than the value itself... I don't believe you need `hash_replace` anymore.  Can we remove that?

Reply:
well, we still need the `hash_replace` if the replace pattern doesn't have a regular expression or the regular expression doesn't have any captured groups. However i'm not sure if we will ever need to hash a static value. Thoughts?

Reply:
I think we should only support hashing captured groups.  It is more in line with the feature requests we see.  Also, if they wish to hash the entire matched expression, they could just surround it with parentheses.